--- grub.default	2019-10-18 13:00:36.855181213 +0100
+++ /etc/default/grub	2019-10-17 02:40:56.088874153 +0100
@@ -8,7 +8,7 @@
 GRUB_HIDDEN_TIMEOUT_QUIET=true
 GRUB_TIMEOUT=10
 GRUB_DISTRIBUTOR=`lsb_release -i -s 2> /dev/null || echo Debian`
-GRUB_CMDLINE_LINUX_DEFAULT="quiet splash"
+GRUB_CMDLINE_LINUX_DEFAULT="efi=novamap"
 GRUB_CMDLINE_LINUX=""
 
 # Uncomment to enable BadRAM filtering, modify to suit your needs
@@ -32,3 +32,5 @@
 
 # Uncomment to get a beep at grub start
 #GRUB_INIT_TUNE="480 440 1"
+
+export GRUB_DTB_PREFIX="/dtb/lenovo-yoga-c630"

